gpu/octree/src/cuda/radius_search.cu

```diff
@@ -254,20 +257,33 @@ namespace pcl
                             __shfl_sync(0xFFFFFFFF, query.z, active_lane)
                         );
 
-                        length = TestWarpKernel(beg, active_query, radius2, length, out, length_left);
+                        length = TestWarpKernel(beg, active_query, radius2, length, out, sqr_dist, length_left);
                     }
                     else
                     {                            
                         length = min(length, length_left);                        
                         Warp::copy(batch.indices + beg, batch.indices + beg + length, out);
+
+                        if (batch.compute_all_distances)
+                        {
+                            //broadcast warp_query
+                            const float3 active_query = make_float3(
+                                __shfl_sync(0xFFFFFFFF, query.x, active_lane),
+                                __shfl_sync(0xFFFFFFFF, query.y, active_lane),
+                                __shfl_sync(0xFFFFFFFF, query.z, active_lane)
+                            );
```

Comment:
Both branches of
```cpp
if(test)
{
  [...]
}
else
{
  [...]
}
```

make use of `active_query`. declare it in the same scope as `test`.